### PR TITLE
Using one connection per puma worker

### LINKED ISSUE #15: Potential leak con connection handling
Currently nuntius is not reusing connections, but re-creating the connection object for each `publish`.

### PR DESCRIPTION
**Pull request by luchotc**
Fixes #15

### CODE REVIEW

File: lib/mumukit/nuntius/connection.rb [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
-      [connection, channel, exchange]
+      [channel, exchange]
+    end
+
+    def establish_connection
+      @@connection = Bunny.new(host: config[:host], user: config[:user], password: config[:password])
+    end
+
+    def connection
+      @@connection ||= nil
```
Comment (flbulgarelli) [inline lib/mumukit/nuntius/connection.rb]: The `||= nil` is not necessary. `nil` is the default value.

File: lib/mumukit/nuntius/publisher.rb [hunk 2]
```diff
@@ -3,9 +3,9 @@ module Mumukit::Nuntius::Publisher
   class << self
 
     def publish(exchange_name, data, opts={})
-      connection, channel, exchange = Mumukit::Nuntius::Connection.start(exchange_name)
```
Comment (flbulgarelli) [inline lib/mumukit/nuntius/publisher.rb]: Perhaps we should rename this class to `Channel`

File: lib/mumukit/nuntius/connection.rb (hunk 1 above)
Comment (luchotc) [reply lib/mumukit/nuntius/connection.rb]: This avoids checking if the class variable is defined.
I could just use the class variable and ask for definition if you prefer

Comment (flbulgarelli) [reply lib/mumukit/nuntius/connection.rb]: OK, but why aren't we using a plain class-instance variable? `@connection` would just work, right?

File: lib/mumukit/nuntius/publisher.rb (hunk 2 above)
Comment (luchotc) [reply lib/mumukit/nuntius/publisher.rb]: I'm not sure if we should rename the class Channel, or just rename the method start to `start_channel`. The method config still refers to the connection

**Pull request merged**